macros/edr/tests/test_collect_metrics.sql

```diff
@@ -0,0 +1,67 @@
+{% test collect_metrics(
+    model,
+    metrics=none,
+    timestamp_column=none,
+    time_bucket=none,
+    days_back=64,
+    backfill_days=none,
+    where_expression=none,
+    dimensions=none
+) %}
+
+    {{ config(
+        tags=['elementary-tests'],
+        meta={"elementary": {"include": false}}
+    ) }}
+
+    {% if not execute or not elementary.is_test_command() or not elementary.is_elementary_enabled() %}
+        {% do return(elementary.no_results_query()) %}
+    {% endif %}
+
+    {% do elementary.debug_log("Metrics: {}".format(metrics)) %}
+
+    {% if not dimensions %}
+        {% set dimensions = [] %}
+    {% endif %}
+
+    {% set model_relation = elementary.get_model_relation_for_test(model, context["model"]) %}
+    {% if not model_relation %}
+        {% do exceptions.raise_compiler_error("Unsupported model: " ~ model ~ " (this might happen if you override 'ref' or 'source')") %}
+    {% endif %}
+
+    {% set table_metrics = [] %}
+    {% set col_to_metrics = {} %}
+    {% for metric in metrics %}
+        {% if metric.get("columns") %}
```

Comment:
Maybe let's add a strict validation on which metrics must have columns and which must not?

E.g. I think you should get an explicit error if you're adding a `row_count` column metric, or a `null_count` table metric.